File: SiliconLabsApp/ViewControllers/AppSelection/SILDevelopNavigationViewController.m
//
//  SILDevelopNavigationViewController.m
//  BlueGecko
//
//  Created by Kamil Czajka on 24/01/2020.
//  Copyright © 2020 SiliconLabs. All rights reserved.
//

#import "SILDevelopNavigationViewController.h"

@interface SILDevelopNavigationViewController () <UINavigationControllerDelegate>

@end

@implementation SILDevelopNavigationViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.delegate = self;
    // Do any additional setup after loading the view.
}

- (void)navigationController:(UINavigationController *)navigationController willShowViewController:(UIViewController *)viewController animated:(BOOL)animated {
    if ([viewController isKindOfClass:[SILAppSelectionViewController class]]) {
        SILAppSelectionViewController* developController = viewController;
        developController.appsArray = [SILApp developApps];
        developController.isDisconnectedIntentionally = NO;
    }
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: SiliconLabsApp/ViewControllers/AppSelection/SILDevelopNavigationViewController.swift
//
//  SILDevelopNavigationViewController.swift
//  BlueGecko
//
//  Created by Grzegorz Janosz on 13/05/2021.
//  Copyright © 2021 SiliconLabs. All rights reserved.
//

import Foundation

class SILDevelopNavigationViewController: UINavigationController, UINavigationControllerDelegate {
    
    override func viewDidLoad() {
        super.viewDidLoad()
        delegate = self
    }

    func navigationController(_ navigationController: UINavigationController, willShow viewController: UIViewController, animated: Bool) {
        if let developController = viewController as? SILAppSelectionViewController, let developApps = SILApp.developApps() as? [SILApp] {
            developController.appsArray = developApps
            developController.isDisconnectedIntentionally = false
            let tabBar = self.tabBarController?.tabBar as? SILTabBar
            tabBar?.setMuliplierForSelectedIndex(1)
        }
    }
}
